experimental/sql/src/projection_pushdown.py

```diff
@@ -84,12 +128,26 @@ def match_and_rewrite(self, op: RelAlg.Operator, rewriter: PatternRewriter):
     if None in cols:
       return
 
-    new_proj = RelAlg.Project.get(
-        Region.from_operation_list([op.input.op.clone()]),
-        Region.from_operation_list([RelAlg.Column.get(s) for s in cols]),
-        ArrayAttr.from_list([StringAttr.from_str(s) for s in cols]))
-
-    rewriter.replace_op(op.input.op, new_proj)
+    if not isinstance(op, RelAlg.CartesianProduct):
```

Comment:
Isn't is more future-proof to do it the other way around? Handle the `isinstance(op, CartesianProduct)` case first, then handle other special cases in the future, and finally handle the default case?